### PR TITLE
Corrects SecDebugLogLevel doc, reorders logs levels

### PR DESCRIPTION
**Pull request by M4tteoP**
Hello! I'm trying to reconcile the log-level implementation with the information provided inside `coraza.conf-recommended`.
This is the summary of the current situation ([ref](https://github.com/corazawaf/coraza/issues/252#issuecomment-1263777437)):
| [coraza.conf-recommended](https://github.com/corazawaf/coraza/blob/5408fcdf553217ea45ec9abda978e3fa593e663b/coraza.conf-recommended#L206-L217)  | Documentation | [Current implementation](https://github.com/corazawaf/coraza/blob/fce26f25ab3e1955de7b396d01d6b43dc8c33b7b/loggers/debug.go#L33) (v3/dev) |
| ------------- | ------------- | ------------- |
| 1: fatal<br>2: panic<br>3: error<br>4: warn<br>5: info<br>6: debug | 0: Fatal<br>1: Panic<br>2: Error<br>3: Warning<br>4: details of how transactions are handled<br>5: log everything |0: LogLevelUnknown<br>1: LogLevelInfo<br>2: LogLevelWarn<br>3: LogLevelError<br>4: LogLevelDebug<br>5: LogLevelTrace  |

The PR:
- Reorders the log levels from most verbose to least one in the following way: trace -> debug -> info -> warning -> error
- Updates the recommended file showing only these five valid values. 

I'm wondering:
- What was the difference between `Fatal`, `Panic` and `Error`? Zap heritage?
- Do we want to provide a ~~`LogLevelFatal`~~ `LogLevelNoLog` that basically is a no-log level? (it would be at `0`).
- What is the reasoning behind `info` log level? The implementation considers it the least verbose level but `recommended` shows it as one of the most verbose ones. The PR places it in an intermediate verbosity, like ModSec behaviour. 

Closes https://github.com/corazawaf/coraza/issues/252.

### CODE REVIEW

**Comment by fzipi** [comment]
I don't think naming `LogLevelFatal` something that doesn't log makes any sense 🤔 At the bare minimum, it's confusing.

**Comment by jcchavezs** [comment]
Naive question: shall we follow modsec levels https://www.feistyduck.com/library/modsecurity-handbook-free/online/ch04-logging.html?

0 | No log | No log
1 | Errors (e.g., fatal processing errors, blocked transactions) | Fatal
2 | Warnings (e.g., non-blocking rule matches) | Warn 
3 | Notices (e.g., non-fatal processing errors) | Error
4 | Informational | info
5 | Detailed | Debug
9 | Everything! | Trace

?

**Comment by jptosso** [comment]
What are we going to do about this?

I think we should keep apache error levels for compatibility, but fatal to error should default to error

**Comment by jcchavezs** [comment]
@M4tteoP is going to come up with another alternative.

**Comment by M4tteoP** [comment]
According to the docs, ModSecurity levels for both v2 and v3 are the following:
```
0: no logging
1: errors (intercepted requests) only
2: warnings
3: notices
4: details of how transactions are handled
5: as above, but including information about each piece of information handled
9: log everything, including very detailed debugging information
```
Therefore, I came up with the following correspondence in Coraza:
```
0: no logging
1: error
2: warnings
3: info
4-5: debug
6-9: trace
```
The last commit:
- Adds the `NoLogLevel`
- Sets "official" Coraza log levels to the lower correct number, therefore no changes on loggers implementations are needed. (E.g. Debug is set with `5`, it will be `>` than `DebugLogLevel` (`4`), therefore it will be printed.

Let me know if it seems right, or if you were expecting another logic.

**Review by jcchavezs — APPROVED** [review]

**Review by fzipi — APPROVED** [review]
Amazing work here!

**Review by anuraaga — APPROVED** [review]

File: coraza.conf-recommended [hunk 1]
```diff
@@ -205,12 +205,12 @@ SecDataDir /tmp/
...
-# 3: error
-# 4: warn
-# 5: info
-# 6: debug
+# 0: no logging (least verbose)
+# 1: error
+# 2: warn
+# 3: info
+# 4-5: debug
+# 6-9: trace (most verbose)
```
Comment (jcchavezs) [inline coraza.conf-recommended:213]: Is this still correct? Isn't it the cas that 4-8 is debug?

Comment (M4tteoP) [reply coraza.conf-recommended:213]: 🤦🏻‍♂️

**Pull request merged**